Snabble/UI/ShoppingCart/ShoppingCartViewController.swift

```diff
@@ -470,9 +484,15 @@ public final class ShoppingCartViewController: UIViewController {
                         return
                     }
 
-                    if error.error.type == .noAvailableMethod {
+                    switch error.error.type {
+                    case .noAvailableMethod:
                         self.delegate?.showWarningMessage("Snabble.Payment.noMethodAvailable".localized())
                         return
+                    case .invalidDepositVoucher:
+                        self.delegate?.showWarningMessage("Snabble.invalidDepositVoucher.errorMsg".localized())
+                        return
+                    default:
+                        ()
                     }
 
                     // app didn't handle the error. see if the project has a offline-capable payment method
```

Comment:
Den Teil könnten wir komplett in das Switch-Case ziehen:

1. `return` innerhalb der Cases können entfernt werden
2. `default` ist befüllt, da dies genau der Fall ist _"app didn't handle the error..."_

```swift
                    switch error.error.type {
                    case .noAvailableMethod:
                        self.delegate?.showWarningMessage("Snabble.Payment.noMethodAvailable".localized())
                    case .invalidDepositVoucher:
                        self.delegate?.showWarningMessage("Snabble.invalidDepositVoucher.errorMsg".localized())
                    default:
                        if !offlineMethods.isEmpty {
                            let info = SignedCheckoutInfo(offlineMethods)
                            self.delegate?.gotoPayment(paymentMethod, self.methodSelector?.selectedPaymentDetail, info, self.shoppingCart)
                        } else {
                            self.delegate?.showWarningMessage("Snabble.Payment.errorStarting".localized())
                        }
                    }
```

Was hältst du davon? Empfinde ich wesentlich verständlich, als mit einem Kommentar den Default Fall zu deklarieren und zwei zusätzliche `return`.

Reply:
stimmt, ist besser.